sympy/matrices/tests/test_decompositions.py

```diff
@@ -303,7 +303,7 @@ def goofyiszero(value):
         assert magic_string == err.args[0]
         return
 
-    assert False
+    raise AssertionError()
```

Comment:
I think `assert False` does not need to be changed in testing code. The general argument against using assert False is that asserts can be disabled by running python with `-O`. The entire sympy test suite depends on assertions not being disabled though.

Reply:
Yeah, but it's still better behavior. Assert False just raises an AssertionError behind the scenes anyway.

Reply:
Exactly so they are equivalent but `assert False` is the standard spelling. In principle there are reasons not to use it in main code but those don't apply to test code.

I think that probably there are better ways to spell all of these than using either `assert False` or `raise AssertionError`. This particular case for example is just unreachable code that can be deleted.

This whole test actually seems bogus to me but maybe it would have prevented the related bug that is seen for `LUsolve`:
https://github.com/sympy/sympy/issues/24679#issuecomment-1468017766. A better test would be that iszerofunc actually does something useful.

Reply:
This should just be deleted.